llvm/unittests/Target/AIE/AIEScheduleDAGMITest.cpp

```diff
@@ -61,4 +66,26 @@ TEST_F(AIEScheduleDAGMITest, SchedNoDelta) {
   EXPECT_EQ(MISeq(*MBB), MISeq({MI1, MI0, MI2}));
 }
 
+/// Case where instructions are scheduled with a delta from Bot.getCurrCycle().
+TEST_F(AIEScheduleDAGMITest, SchedWithDelta) {
+  auto *MI0 = appendPlainInstr();
+  auto *MI1 = appendPlainInstr();
+  auto *MI2 = appendPlainInstr();
+
+  initializeScheduler();
+  SchedBoundary &Bot = Scheduler->getSchedZone();
+
+  // Mark all instructions as available.
+  for (auto *MI : {MI0, MI1, MI2})
+    Bot.releaseNode(Scheduler->getSUnit(MI), /*ReadyCycle=*/0, false);
+
+  Scheduler->scheduleInstr(MI2, Bot, -8); // Emit in cycle 0+8
+  Bot.bumpCycle(2);
+  Scheduler->scheduleInstr(MI1, Bot, -5); // Emit in cycle 2+5
+  Bot.bumpCycle(3);
+  Scheduler->scheduleInstr(MI0, Bot); // Emit in cycle 3
```

Comment:
CHECK, we can not call scheduleInstr out of order

Reply:
Not sure what you mean by out-of-order. Here we are scheduling bottom-up, but the first instruction is scheduled in cycle 8, while the next ones in cycle 7 and 3. Meaning they are "below" in the final schedule. I think that somewhat counts as out-of-order, given that we are scheduling from the bottom, but the instructions actually appear to be emitted top-down.